vpc/service/auth.go

```diff
@@ -57,12 +58,37 @@ func (vpcService *titusVPCAgentServiceAuthFuncOverride) AuthFuncOverride(ctx con
 	if err != nil {
 		return ctx, errors.Wrap(err, "Cannot parse remote certificate")
 	}
-	_, err = cert.Verify(x509.VerifyOptions{
+	err = vpcService.validateCert(cert)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to verify client cert")
+	}
+	return ctx, nil
+}
+
+func (vpcService *titusVPCAgentServiceAuthFuncOverride) validateCert(cert *x509.Certificate) error {
+	// This first check only validates the chain of trust
+	_, err := cert.Verify(x509.VerifyOptions{
 		Roots:     vpcService.TitusAgentCACertPool,
 		KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "Unable to verify client cert")
+		return err
 	}
-	return ctx, nil
+	// The cert.Verify function is relativly limited in what it can verify.
+	// It only can take in a single string and match it against DNS/CN.
+	// We know we will need to match against a variety of incoming names,
+	// so we check everything in a loop and return an error if nothing matched.
+	validCNs := []string{`titusagent\..*`}
```

Comment:
Please make this a configuration value.

Reply:
Added `SSL_VALID_CNS` and the flag for it.

I feel a little weird doing a csv? But in the end strings are kinda all we have for input config this way. Hopefully we will very rarely need to change this away from the default.

Reply:
A CSV? What do you mean a CSV?

Reply:
comma separated value?

I know it is lame. Do you think it would be better to just have 1 regex only and leave it as an exercise to the reader to craft a regex that meets all their needs?

Writing this now it seems kinda obvious we should just have 1 regex and not an array of them...

Reply:
Just add a command to combine regexes:

https://github.com/spadgos/regex-combiner#:~:text=Combines%20an%20array%20of%20regexes%20into%20a%20single%20regex.

Reply:
(yes, there should be 1 regex)